cmd/oras/root/manifest/fetch.go

```diff
@@ -91,15 +91,19 @@ Example - Fetch raw manifest from an OCI layout archive file 'layout.tar':
 	return cmd
 }
 
-func fetchManifest(ctx context.Context, opts fetchOptions) (fetchErr error) {
+func fetchManifest(ctx context.Context, cmd *cobra.Command, opts fetchOptions) (fetchErr error) {
 	ctx, logger := opts.WithContext(ctx)
 
 	target, err := opts.NewReadonlyTarget(ctx, opts.Common, logger)
 	if err != nil {
 		return err
 	}
 	if err := opts.EnsureReferenceNotEmpty(); err != nil {
-		return err
+		return &oerrors.Error{
```

Comment:
This enhance should be built into the `EnsureReferenceNotEmpty` so other commands (e.g. discover, pull, tag...) can have this error output improvement.

Reply:
This function is used in 10 commands (see changed files). We probably need to customize the error message for `cp`, `tag` and maybe `blob fetch/delete`.